sbi/inference/snle/snle_a.py

```diff
@@ -51,53 +51,3 @@ def __init__(
 
         kwargs = del_entries(locals(), entries=("self", "__class__"))
         super().__init__(**kwargs)
-
-    def train(
-        self,
-        training_batch_size: int = 50,
-        learning_rate: float = 5e-4,
-        validation_fraction: float = 0.1,
-        stop_after_epochs: int = 20,
-        max_num_epochs: int = 2**31 - 1,
-        clip_max_norm: Optional[float] = 5.0,
-        exclude_invalid_x: bool = True,
-        resume_training: bool = False,
-        discard_prior_samples: bool = False,
-        retrain_from_scratch: bool = False,
-        show_train_summary: bool = False,
-        dataloader_kwargs: Optional[Dict] = None,
-    ) -> flows.Flow:
-        r"""Return density estimator that approximates the distribution $p(x|\theta)$.
-
-        Args:
-            training_batch_size: Training batch size.
-            learning_rate: Learning rate for Adam optimizer.
-            validation_fraction: The fraction of data to use for validation.
-            stop_after_epochs: The number of epochs to wait for improvement on the
-                validation set before terminating training.
-            max_num_epochs: Maximum number of epochs to run. If reached, we stop
-                training even when the validation loss is still decreasing. Otherwise,
-                we train until validation loss increases (see also `stop_after_epochs`).
-            clip_max_norm: Value at which to clip the total gradient norm in order to
-                prevent exploding gradients. Use None for no clipping.
-            exclude_invalid_x: Whether to exclude simulation outputs `x=NaN` or `x=±∞`
-                during training. Expect errors, silent or explicit, when `False`.
-            resume_training: Can be used in case training time is limited, e.g. on a
-                cluster. If `True`, the split between train and validation set, the
-                optimizer, the number of epochs, and the best validation log-prob will
-                be restored from the last time `.train()` was called.
-            discard_prior_samples: Whether to discard samples simulated in round 1, i.e.
-                from the prior. Training may be sped up by ignoring such less targeted
-                samples.
-            retrain_from_scratch: Whether to retrain the conditional density
-                estimator for the posterior from scratch each round.
-            show_train_summary: Whether to print the number of epochs and validation
-                loss and leakage after the training.
-            dataloader_kwargs: Additional or updated kwargs to be passed to the training
-                and validation dataloaders (like, e.g., a collate_fn)
-
-        Returns:
-            Density estimator that approximates the distribution $p(x|\theta)$.
-        """
-        kwargs = del_entries(locals(), entries=("self", "__class__"))
-        return super().train(**kwargs)
```

Comment:
In the PR description you write `MNLE is a child of SNLE and can be used either by passing model="mnle"....`. Is this indeed possible? (I'm fine with not having this feature, but please remove it from the PR description)

Reply:
yes, in principle this is possible, but it will result in a posterior that is slower because it does not optimize the evaluation for mixed data. I think it is rather unlikely that this is used at all, but should we still issue a warning for this?

Reply:
thinking about it again it appears like a corner case unlikely to happen, and it would require issue a warning about `mnle` in the base class which is not elegant. I decided to remove it from the description and just leave it like it is.

Reply:
I agree. Makes sense!